activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -1,32 +1,48 @@
 module ActiveSupport
   module Testing
     class SimpleStubs # :nodoc:
-      Stub = Struct.new(:object, :method_name, :original_method)
+      Stub = Struct.new(:object, :method_name, :original_method, :return_value)
 
       def initialize
-        @stubs = {}
+        @stubs = Hash.new { |h,k| h[k] = [] }
       end
 
       def stub_object(object, method_name, return_value)
         key = [object.object_id, method_name]
 
-        if stub = @stubs[key]
-          unstub_object(stub)
+        if @stubs[key].any?
+          stub = @stubs[key].last
+          unstub_object(stub) if stub
         end
 
         new_name = "__simple_stub__#{method_name}"
 
-        @stubs[key] = Stub.new(object, method_name, new_name)
+        @stubs[key].push(Stub.new(object, method_name, new_name, return_value))
 
         object.singleton_class.send :alias_method, new_name, method_name
         object.define_singleton_method(method_name) { return_value }
       end
 
       def unstub_all!
-        @stubs.each_value do |stub|
-          unstub_object(stub)
+        @stubs.each_value do |stubs|
+          stub = stubs.first
+          unstub_object(stub) if stub
+        end
+        @stubs.clear
+      end
+
+      def unstub_to_previous_stub_state(object, method_name, return_value)
+        key = [object.object_id, method_name]
+
+        if @stubs[key].any?
+          current_stub = @stubs[key].pop
+
+          if stub = @stubs[key].last
+            stub_object(stub.object, stub.method_name, stub.return_value)
+          else
+            unstub_object(current_stub)
+          end
```

Comment:
Could do with a better method name 😁

Since the stubs per method is essentially turned into a stack we could go with `pop_stub`.

Here's my take:

```ruby

def pop_stub(object, method_name)
  key = [ object.object_id, method_name ]
  currently_stubbed_to = @stubs[key].pop
  if stub_next_in_stack = @stubs[key].last
    stub_object(object, method_name, stub_next_in_stack.return_value)
  else
    unstub_object(currently_stubbed_to)
  end
end
```

Renamed some variables because I found them hard to keep track of.